ledger/src/shred.rs

```diff
@@ -811,129 +811,54 @@ impl Shredder {
             .collect()
     }
 
-    fn fill_in_missing_shreds(
-        num_data: usize,
-        num_coding: usize,
-        first_index_in_fec_set: usize,
-        expected_index: usize,
-        index_found: usize,
-        present: &mut [bool],
-    ) -> Vec<Vec<u8>> {
-        let end_index = index_found.saturating_sub(1);
-        // The index of current shred must be within the range of shreds that are being
-        // recovered
-        if !(first_index_in_fec_set..first_index_in_fec_set + num_data + num_coding)
-            .contains(&end_index)
-        {
-            return vec![];
-        }
-
-        let missing_blocks: Vec<Vec<u8>> = (expected_index..index_found)
-            .map(|missing| {
-                present[missing.saturating_sub(first_index_in_fec_set)] = false;
-                if missing < first_index_in_fec_set + num_data {
-                    Shred::new_empty_data_shred().payload
-                } else {
-                    vec![0; SHRED_PAYLOAD_SIZE]
-                }
-            })
-            .collect();
-        missing_blocks
-    }
-
     pub fn try_recovery(
         shreds: Vec<Shred>,
         num_data: usize,
         num_coding: usize,
         first_index: usize,
         slot: Slot,
     ) -> std::result::Result<Vec<Shred>, reed_solomon_erasure::Error> {
+        use reed_solomon_erasure::Error::InvalidIndex;
+        const CODE_BLOCK_SIZE: usize = SHRED_PAYLOAD_SIZE - SIZE_OF_CODING_SHRED_HEADERS;
```

Comment:
nit: maybe a comment here about what this represents, i.e.

"the first `N` bytes of the payload in the data shreds represent the original symbols input into the erasure scheme"

Reply:
moved this further down and added comments.